engine/source/public/core/engine.hpp

```diff
@@ -1,8 +1,14 @@
 ﻿#pragma once
 
+#include "types.hpp"
+
 class Engine
 {
 public:
+    int2 windowExtent {1024, 576};
+    struct SDL_Window* window {nullptr};
```

Comment:
Can we use a smart pointer here instead?

Reply:
no? i can't get it to work. i keep getting the error that the type is incomplete, even tho the type is included

Reply:
This is fine then